RecoHGCal/TICL/python/iterativeTICL_cff.py

```diff
@@ -22,14 +22,20 @@
 pfTICL = _pfTICLProducer.clone()
 ticlPFTask = cms.Task(pfTICL)
 
-iterTICLTask = cms.Task(ticlLayerTileTask
-    ,ticlTrkEMStepTask
+ticlIterations = cms.Task(
+    ticlTrkEMStepTask
     ,ticlEMStepTask
     ,ticlTrkStepTask
     ,ticlHADStepTask
+)
+ticlIterLabels = [_step.itername.value() for _iteration in ticlIterations for _step in _iteration if (_step._TypedParameterizable__type == "TrackstersProducer")]
+
```

Comment:
This approach seems to be rather fragile. In fact, you need to inject manually a few lines below, the `Merge` label that is coming from a different `C++` producer (`TracksterMergeProducer`).
Since the plan is to use the single `Merge` collection and filter the `tracksters` based on their `iteration enum` in the validation, does it make sense to have this PR at all? At least for the `inputTag` part, this would not be needed any-longer. For the folder structure, IIRC, the search could be done recursively starting from the top folder (to be checked).

Reply:
> This approach seems to be rather fragile. In fact, you need to inject manually a few lines below, the `Merge` label that is coming from a different `C++` producer (`TracksterMergeProducer`).

The `TracksterMergeProducer` has no `itername` attribute. I can think of adding it to avoid the manual injection.

> Since the plan is to use the single `Merge` collection and filter the `tracksters` based on their `iteration enum` in the validation, does it make sense to have this PR at all? At least for the `inputTag` part, this would not be needed any-longer. For the folder structure, IIRC, the search could be done recursively starting from the top folder (to be checked).

I would keep this PR code until we figure out how to fully switch to the `iteration enum`.

Reply:
Even if you add it, you would need also to add another case in the `if` on the `C++` type, which would make the code more fragile, I suspect.